src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -27,27 +29,39 @@ namespace NuGet.SolutionRestoreManager
     [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
     [ProvideAutoLoad(VSConstants.UICONTEXT.SolutionExists_string)]
     [Guid(PackageGuidString)]
-    public sealed class RestoreManagerPackage : AsyncPackage
+    public sealed class RestoreManagerPackage : AsyncPackage, IVsUpdateSolutionEvents4
     {
         public const string ProductVersion = "4.0.0";
 
         /// <summary>
+        /// Wait timeout for to acquire NuGet lock during build event. Don't block build if we can't 
+        /// acquire lock.
+        /// </summary>
+        private const int NuGetLockTimeout = 0;
```

Comment:
I appreciate the xml doc comment on this member, but it doesn't describe the units (is this ms?) nor how `0` is interpreted (wait indefinitely, or wait for no time at all?).